common/src/main/scala/uk/ac/wellcome/platform/archive/common/storage/services/TagManifestFileFinder.scala

```diff
@@ -52,10 +49,20 @@ class TagManifestFileFinder[IS <: InputStream with HasLength](
   ): Option[StorageManifestFile] =
     streamReader.get(prefix.asLocation(name)) match {
       case Right(is) =>
+        // TODO: we could handle the checksum more gracefully here, not '.get'
+        //
+        // This method is called in the bag register to create the storage manifest,
+        // so it happens after the entire bag has been verified.  To verify the bag,
+        // we've already had to read the tagmanifest-sha256.txt file, so an error
+        // here would be unlikely (but probably not impossible).
+        val checksum = Hasher
+          .hash(is.identifiedT)
+          .get
```

Comment:
Can we at least catch and re-throw with a more sensible error?

Reply:
Is good suggestion, will change.